src/components/MolgenisTable.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <table :class="{
+    'mg-normal-font': needsNormalFont,
+    'table-sm': needsSmallerPadding,
+    'mg-mid-font': needsMidFont,
+    'mg-min-padding': needsMinPadding,
+    'mg-small-font': needsSmallFont,
+    'mg-xs-font': needsXsFont
+  }" class="table table-bordered mg-table" ref="table">
+    <caption>{{title}}</caption>
+    <thead>
+    <tr>
+      <th v-for="(col, index) in header" :key="index" scope="col">{{col}}</th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr v-for="(row, index) in data" :key="index">
+      <td v-for="(value, index) in row" :key="index" v-html="value"/>
+    </tr>
+    </tbody>
+  </table>
+</template>
+
+<script>
+import { isAbove, isBelow, isBetween } from '@/helpers/range'
+
+export default {
+  name: 'molgenis-table',
+  props: { title: String, header: Array, data: Array },
+  data () {
+    return {
+      tableWidth: 0
+    }
+  },
+  methods: {
+    onResize () {
+      this.tableWidth = this.$refs.table.clientWidth
+    }
+  },
+  computed: {
+    dataLength () {
+      return this.data.length
+    },
+    headerLength () {
+      return this.header.length
+    },
+    needsNormalFont () {
+      return !(this.needsMidFont || this.needsSmallFont || this.needsXsFont) && (isBelow(this.headerLength, 8) || isBelow(this.dataLength, 11))
```

Comment:
This looks kinda complex, have you thought about using relatieve font sizes (em/rem) and setting all font scaling by changing the base font size?

Reply:
I don't really understand what you mean by that. I'm not that familiar with css magic. I mean, I have basic knowledge but I'm not all-knowing haha

Reply:
I made a small tutorial to explain. I just realized that is should have made it in molgenis-tutorial. Would have worked perfectly there! 😄 

https://codepen.io/maxpostema/pen/BazRpNm

Reply:
Just realized that bootstrap (4 pages) use this trick. Go to https://master.dev.molgenis.org/ and change the font-size on the HTML tag. The site will scale with the value.

Reply:
Tangent: also check the 'vh' and 'vw' types. they may also help with relative scaling

Reply:
Thanks for the explanation! I did know about bootstrap with the main fontsize on the HTML and using rems to make the fontsize of childobjects based on the base fontsize. I'm just wondering how you would include it in the table object. Using bootstraps scaling isn't enough to let the table shrink once you get more data and having a very small table when you don't have much data doesn't look very appealing. Do you have any suggestions on how to apply it here?